### PR TITLE
Add note about delegation crates to Deref anti-pattern

### PR DESCRIPTION
**Pull request by Kobzol**
Closes https://github.com/rust-unofficial/patterns/issues/192

### CODE REVIEW

File: anti_patterns/deref.md [hunk 1]
```diff
@@ -110,6 +110,10 @@ stable Rust. See these [blog](http://aturon.github.io/blog/2015/09/18/reuse/)
 [posts](http://smallcultfollowing.com/babysteps/blog/2015/10/08/virtual-structs-part-4-extended-enums-and-thin-traits/)
 and this [RFC issue](https://github.com/rust-lang/rfcs/issues/349) for more details.
 
+Until there is direct support for handling delegation in Rust, you can try some crates that help
+reduce method delegation boilerplate, for example [delegate](https://crates.io/crates/delegate)
+or [ambassador](https://crates.io/crates/ambassador).
```
Comment (pickfire) [inline anti_patterns/deref.md:115]:
```suggestion
Until there is direct support for handling delegation in Rust, you can try some 
crates that help reduce method delegation boilerplate, for example
[delegate](https://crates.io/crates/delegate) or
[ambassador](https://crates.io/crates/ambassador).
```
80 characters wide required, thanks for sending the patch.

**Review by pickfire — DISMISSED** [review]
Looks good to me.

**Comment by simonsan** [comment]
Hmm, I would actually just put the two links into the `See also` section, as we did it with other recommendations (i guess)

**Comment by simonsan** [comment]
Like this i mean, what do you both think of it @pickfire & @Kobzol

File: anti_patterns/deref.md [hunk 2]
```diff
@@ -123,6 +123,6 @@ conversion between arbitrary types.
 
 ## See also
 
-[Collections are smart pointers idiom](../idioms/deref.md).
-
-[Documentation for `Deref` trait](https://doc.rust-lang.org/std/ops/trait.Deref.html).
+- [Collections are smart pointers idiom](../idioms/deref.md).
+- Delegation crates for less boilerplate like [delegate](https://crates.io/crates/delegate) or [ambassador](https://crates.io/crates/ambassador)
```
Comment (simonsan) [inline anti_patterns/deref.md:127]:
```suggestion
- Delegation crates for less boilerplate like [delegate](https://crates.io/crates/delegate) or 
  [ambassador](https://crates.io/crates/ambassador)
```

**Comment by Kobzol** [comment]
I think that it's fine, as long as the crates are mentioned, it doesn't matter to me where.

File: anti_patterns/deref.md [hunk 3]
```diff
@@ -123,6 +123,7 @@ conversion between arbitrary types.
 
 ## See also
 
-[Collections are smart pointers idiom](../idioms/deref.md).
-
-[Documentation for `Deref` trait](https://doc.rust-lang.org/std/ops/trait.Deref.html).
+- [Collections are smart pointers idiom](../idioms/deref.md).
+- Delegation crates for less boilerplate like [delegate](https://crates.io/crates/delegate) or 
+  [ambassador](https://crates.io/crates/ambassador)
```
Comment (simonsan) [inline anti_patterns/deref.md:128]:
```suggestion
- Delegation crates for less boilerplate like [delegate](https://crates.io/crates/delegate)
  or [ambassador](https://crates.io/crates/ambassador)
```

**Review by simonsan — APPROVED** [review]

**Comment by simonsan** [comment]
Thank you :)

**Pull request merged**